// @tencent/techmap-api/src/model/MeasureReport.js
/*
 * 技术图谱api文档
 * techmap api
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.19
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Api) {
      root.Api = {};
    }
    root.Api.MeasureReport = factory(root.Api.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The MeasureReport model module.
   * @module model/MeasureReport
   * @version 1.0
   */

  /**
   * Constructs a new <code>MeasureReport</code>.
   * @alias module:model/MeasureReport
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>MeasureReport</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/MeasureReport} obj Optional instance to populate.
   * @return {module:model/MeasureReport} The populated <code>MeasureReport</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('day'))
        obj.day = ApiClient.convertToType(data['day'], 'Number');
      if (data.hasOwnProperty('month'))
        obj.month = ApiClient.convertToType(data['month'], 'Number');
      if (data.hasOwnProperty('targetId'))
        obj.targetId = ApiClient.convertToType(data['targetId'], 'Number');
      if (data.hasOwnProperty('type'))
        obj.type = ApiClient.convertToType(data['type'], 'String');
      if (data.hasOwnProperty('value'))
        obj.value = ApiClient.convertToType(data['value'], 'Number');
      if (data.hasOwnProperty('year'))
        obj.year = ApiClient.convertToType(data['year'], 'Number');
    }
    return obj;
  }

  /**
   * @member {Number} day
   */
  exports.prototype.day = undefined;

  /**
   * @member {Number} month
   */
  exports.prototype.month = undefined;

  /**
   * @member {Number} targetId
   */
  exports.prototype.targetId = undefined;

  /**
   * @member {String} type
   */
  exports.prototype.type = undefined;

  /**
   * @member {Number} value
   */
  exports.prototype.value = undefined;

  /**
   * @member {Number} year
   */
  exports.prototype.year = undefined;


  return exports;

}));
