// @tencent/mk-api/lib/api/search/types.js
import { StringDTO, NumberDTO } from '../../define/base';
import { CoteriesListItem } from '../coteries/baseType';
import { TypeWrapper } from '../../http';
import { pick } from '../../define/helper';
import { UserShortInfo } from '../user/types';
export const SimpleSearch = {
    id: NumberDTO,
    title: StringDTO,
};
export const SimpleSearchHttp = TypeWrapper(SimpleSearch);
export const ISearch = {
    coteries: [pick(CoteriesListItem, ['id', 'name', 'avatar', 'createdAt', 'stats'])],
    isearchResponse: {
        contents: [{
                author: StringDTO,
                content_type: StringDTO,
                count_collection: NumberDTO,
                count_comment: NumberDTO,
                count_like: NumberDTO,
                count_view: NumberDTO,
                create_time: StringDTO,
                modify_time: StringDTO,
                fragment: StringDTO,
                href: StringDTO,
                id: StringDTO,
                title: StringDTO,
            }],
        total: NumberDTO,
    },
    users: [pick(UserShortInfo, ['username', 'name', 'org'])],
};
export const ISearchHttp = TypeWrapper(ISearch);
