// @tencent/mk-api/lib/api/subject/types.js
import { NumberDTO, StringDTO, BaseUser, BaseTime, BooleanDTO, } from '../../define/base';
import { extend, optional, pick } from '../../define/helper';
import { UserShortInfo } from '../user/types';
import { TypeWrapper } from '../../http';
import { Idea } from '../idea/types';
import { AdvertBase } from '../banner/types';
export const SubjectBase = extend({
    id: NumberDTO,
    title: StringDTO,
    description: StringDTO,
    referCount: NumberDTO,
    starCount: NumberDTO,
    pv: NumberDTO,
    source: StringDTO,
    sticked: BooleanDTO,
}, BaseUser, BaseTime);
export const SubjectBaseHttp = TypeWrapper(SubjectBase);
export const Subject = extend({
    curStar: BooleanDTO,
    author: UserShortInfo,
}, optional({
    subjectLink: {
        linkableId: NumberDTO,
    },
}), SubjectBase);
export const SubjectHttp = TypeWrapper(Subject);
export const SubjectWithIdea = {
    subject: [Subject],
    ideas: [pick(Idea, ['contentMd', 'id'])],
};
export const SubjectWithIdeaHttp = TypeWrapper(SubjectWithIdea);
export const SubjectAdvert = extend({
    related: SubjectBase,
}, AdvertBase);
export const SubjectAdvertHttp = TypeWrapper(SubjectAdvert);
