// @tencent/mk-api/lib/api/banner/index.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { AdvertBaseHttp, HomeAdvertHttp, ProjectAdvertHttp, ArticleAdvertHttp } from './types';
import { injectAPIGroup } from '../../inject';
export const MkBannerApi = injectAPIGroup({
    /**
     * 获取问答页广告
     *
     * @returns
     */
    getQaBanner(params) {
        const _a = params || {}, { timeValid = true } = _a, rst = __rest(_a, ["timeValid"]);
        return AdvertBaseHttp.get('/posters/20/position', {
            params: Object.assign({ group_by: 20, position: 201, status: 'active', client: 'all', time_valid: timeValid }, rst),
        });
    },
    /**
     * 获取项目广告
     * @deprecated
     * @returns
     */
    getProjectBanner(params) {
        const _a = params || {}, { timeValid = true } = _a, rst = __rest(_a, ["timeValid"]);
        console.error('开源广场即将下线, 如有调用相关API请尽快删除');
        return ProjectAdvertHttp.get('/posters/30/position', {
            params: Object.assign({ group_by: 30, position: 302, status: 'active', client: 'all', time_valid: timeValid }, rst),
        });
    },
    /**
     * 获取首页广告
     *
     * @returns
     */
    getHomeBanner() {
        return HomeAdvertHttp.list().get('/home/poster_101');
    },
    /**
     * 获取比赛广告
     *
     * @returns
     */
    getContestBanner() {
        return AdvertBaseHttp.list().get('/posters/60/position', {
            params: {
                group_by: 60,
                position: 601,
            },
        });
    },
    /**
     * 获取代码笔记首页广告
     *
     * @returns
     */
    getArticleBanner() {
        return ArticleAdvertHttp.list().get('/posters/70/position', {
            params: {
                group_by: 70,
                position: 701,
            },
        });
    },
});
