// @tencent/mk-api/lib/api/banner/types.js
import { StringDTO, NumberDTO } from '../../define/base';
import { TypeWrapper } from '../../http';
import { ProjectShort } from '../project/types';
import { Answer } from '../answer/types';
import { extend, optional, pick } from '../../define/helper';
import { UserShortInfo } from '../user/types';
export const AdvertBase = {
    // client: StringDTO,
    id: StringDTO,
    type: StringDTO,
    url: StringDTO,
    title: StringDTO,
    banner: StringDTO,
    data1: StringDTO,
    data2: StringDTO,
    createdAt: StringDTO,
};
export const AdvertBaseHttp = TypeWrapper(AdvertBase);
export const ProjectAdvert = extend(AdvertBase, optional({
    related: ProjectShort,
}));
export const ProjectAdvertHttp = TypeWrapper(ProjectAdvert);
export const HomeAdvert = extend(AdvertBase, optional({
    related: extend({
        id: NumberDTO,
        title: NumberDTO,
    }, optional({
        answers: pick(Answer, ['author', 'contentPlain']),
    })),
}));
export const HomeAdvertHttp = TypeWrapper(HomeAdvert);
// 文章广告
export const ArticleAdvert = extend(AdvertBase, {
    related: {
        id: NumberDTO,
        title: NumberDTO,
        readCount: NumberDTO,
        author: UserShortInfo,
    },
});
export const ArticleAdvertHttp = TypeWrapper(ArticleAdvert);
