// @tencent/mk-api/lib/api/message/index.js
import { Http } from '../../http';
import { injectDefaultParams } from '../../base';
import { MessageHttp } from './types';
import { injectAPIGroup } from '../../inject';
export const MkMessageApi = injectAPIGroup({
    /**
     * 根据不同内容获取消息列表
     * all: 所有  comment: 回答与评论  system: 系统消息 action: 点赞或关注
     *
     * @param {('all' | 'comment' | 'system' | 'action')} [type='all']
     * @param {PaginationParams} [params]
     * @returns
     */
    getByType(type, params) {
        return MessageHttp.list().get(`/user/msg_records/${type}`, {
            params: injectDefaultParams(params),
        });
    },
    /**
     * 获取各种消息未读情况
     * all: 所有消息数
     * comment: 评论消息数
     * refer: '@'的消息数
     * @returns
     */
    getStats() {
        return Http.get('/user/msg_records/stats');
    },
    /**
     * 获取各种消息未读情况(移动端使用)
     * @returns
     */
    getMobileStats() {
        return Http.get('/user/msg_records/stats/mobile');
    },
    /**
     * 获取未读消息总数
     *
     * @returns
     */
    getUnreadCount() {
        return Http.get('/user/msg_records/stats_unread');
    },
    /**
     * 设置已读
     *
     * @param {number} id
     * @returns
     */
    setRead(id) {
        return Http.post(`/user/msg_record/${id}/read`);
    },
    /**
     * 全部已读
     *
     * @returns
     */
    setReadAll() {
        return Http.post('/user/msg_records/read_all');
    },
});
