// @tencent/mk-api/lib/api/home/index.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { injectDefaultParams } from '../../base';
import { HomeOutboxHttp, Home24OutboxHttp, HomeStarOutBoxHttp, HomeCoteriesListHttp, HomeRecommendUserHttp, } from './types';
import { injectAPIGroup } from '../../inject';
export const MkHomeApi = injectAPIGroup({
    /**
     * 关注列表， 用户关注的人及标签等等最新信息
     *
     * @param {PaginationParams} [params]
     * @returns
     */
    getMoments(params) {
        return HomeStarOutBoxHttp.list().get('/home/star_2', {
            params: injectDefaultParams(params),
        });
    },
    /**
     * 用户关注人/标签发起的回答/提问
     * 比上面的接口轻量
     *
     * @param {PaginationParams} [params]
     * @returns
     */
    getMomentsLite(params) {
        return HomeStarOutBoxHttp.list().get('/home/star_3', {
            params: injectDefaultParams(params),
        });
    },
    /**
     * 推荐列表
     *
     * @param {PaginationParams} [params]
     * @returns
     */
    getRecommend(params) {
        return HomeOutboxHttp.get('/home/recommend', {
            params: injectDefaultParams(params),
        });
    },
    /**
     * 获取24小时动态
     *
     * @param {Get24hoursActivityParams} params
     * @returns
     */
    get24hoursActivity(params) {
        const { afterAt, beforeAt } = params, rst = __rest(params, ["afterAt", "beforeAt"]);
        return Home24OutboxHttp.list().get('/home/24_hours', {
            params: injectDefaultParams(Object.assign(Object.assign({}, rst), { after_at: afterAt, before_at: beforeAt })),
        });
    },
    /**
     * 首页获取用户加入的圈子列表
     *
     * @param {string} username
     * @returns
     */
    getCoteries(username) {
        return HomeCoteriesListHttp.list().get(`/user/${username}/coteries/user_join/home`);
    },
    /**
     * 获取推荐关注人
     *
     * @returns
     */
    getRecommendUser() {
        return HomeRecommendUserHttp.get('/home/recommend_user');
    },
});
