// @tencent/mk-api/lib/api/coteries/baseType.js
import { BaseTime, StringDTO, NumberDTO, KV, BooleanDTO, ShortKV, } from '../../define/base';
import { extend, pick } from '../../define/helper';
// 圈子数据类型
export const CoterieStats = {
    id: NumberDTO,
    coterieId: NumberDTO,
    ideaCount: NumberDTO,
    memberCount: NumberDTO,
    problemCount: NumberDTO,
    articleCount: NumberDTO,
    answeredProblemCount: NumberDTO,
    pv: NumberDTO,
    lastWeekIdeaCount: NumberDTO,
    lastWeekProblemCount: NumberDTO,
    lastWeekMemberCount: NumberDTO,
    lastWeekPv: NumberDTO,
    lastWeekWeight: NumberDTO,
    lastWeekRanking: NumberDTO,
};
// 圈子配置
export const CoterieConfig = {
    id: NumberDTO,
    // coterieId: NumberDTO,
    tags: [KV],
    showInHomePage: BooleanDTO,
    projects: [extend(KV, { extend: StringDTO })],
    oteams: [KV],
    homepageMd: StringDTO,
    homepage: StringDTO,
    monthlyHonors: [KV],
    msgEnabled: BooleanDTO,
    tapdEnabled: BooleanDTO,
};
// 用于获取圈子基本信息
export const CoterieBase = extend({
    id: NumberDTO,
    name: StringDTO,
    avatar: StringDTO,
    status: StringDTO,
    v: StringDTO,
    categories: [ShortKV],
    description: StringDTO,
    stats: pick(CoterieStats, [
        'memberCount',
        'problemCount',
        'ideaCount',
        'lastWeekWeight',
        'lastWeekRanking',
        'lastWeekPv',
        'pv',
    ]),
}, BaseTime);
// 用于获取圈子详情
export const CoterieDetail = extend(CoterieBase, {
    curJoin: BooleanDTO,
    curStar: BooleanDTO,
    stats: pick(CoterieStats, [
        'memberCount',
        'problemCount',
        'articleCount',
        'ideaCount',
        'lastWeekWeight',
        'lastWeekRanking',
        'answeredProblemCount',
        'lastWeekPv',
        'pv',
    ]),
    config: pick(CoterieConfig, [
        'oteams',
        'projects',
        'showInHomePage',
        'tags',
        'homepageMd',
        'homepage',
        'monthlyHonors',
        'msgEnabled',
        'tapdEnabled',
    ]),
});
// 用于获取圈子列表, 附带基本信息
export const CoteriesListItem = pick(CoterieDetail, [
    'id',
    'name',
    'avatar',
    'stats',
    'curJoin',
    'curStar',
    'description',
    'v',
    'createdAt',
]);
