// @tencent/mk-api/lib/api/tag/types.js
import { NumberDTO, StringDTO, BooleanDTO, } from '../../define/base';
import { TypeWrapper } from '../../http';
import { extend } from '../../define/helper';
export const Tag = {
    id: NumberDTO,
    name: StringDTO,
    type: StringDTO,
    subscribeCount: NumberDTO,
    problemRefCount: NumberDTO,
    articleRefCount: NumberDTO,
    contributors: StringDTO,
    notifications: StringDTO,
    createdAt: NumberDTO,
    curStar: BooleanDTO,
};
export const TagHttp = TypeWrapper(Tag);
export const SubTag = {
    id: NumberDTO,
    name: StringDTO,
    parentId: NumberDTO,
};
const TagWithChildren = extend({
    childTagList: [SubTag],
}, Tag);
export const TagWithChildrenHttp = TypeWrapper(TagWithChildren);
export const TagForSelect = {
    name: StringDTO,
    value: NumberDTO,
    selected: BooleanDTO,
};
export const TagForSelectHttp = TypeWrapper(TagForSelect);
export const TagForRecommend = {
    id: NumberDTO,
    name: StringDTO,
    positionTag: BooleanDTO,
};
export const TagForRecommendHttp = TypeWrapper(TagForRecommend);
