// @tencent/mk-api/lib/api/activity/types.js
import { BooleanDTO, NumberDTO, StringDTO, } from '../../define/base';
import { extend } from '../../define/helper';
import { TypeWrapper } from '../../http';
import { UserShortInfo } from '../user/types';
import { QuestionListItemWithSticked } from '../question/types';
import { ArticleList } from '../article/types';
import { ActivityBase } from './baseType';
export const ActivityBaseHttp = TypeWrapper(ActivityBase);
export const Activity = extend({
    detailMd: StringDTO,
    guests: StringDTO,
    template: StringDTO,
    author: UserShortInfo,
    answerCount: NumberDTO,
    noteCount: NumberDTO,
}, ActivityBase);
export const ActivityHttp = TypeWrapper(Activity);
// 活动管理后台问答
const ActivityAdminQuestions = extend({
    hidden: BooleanDTO,
}, QuestionListItemWithSticked);
export const ActivityAdminQuestionListHttp = TypeWrapper(ActivityAdminQuestions);
// 活动管理后台笔记
const ActivityAdminArticles = extend({
    hidden: BooleanDTO,
}, ArticleList);
export const ActivityAdminArticleListHttp = TypeWrapper(ActivityAdminArticles);
