// @tencent/mk-api/lib/api/magazine/index.js
import { Http } from '../../http';
import { magazineNoteHttp } from './types';
import { NoteBaseHttp } from '../../define/note';
import { injectAPIGroup } from '../../inject';
export const MkMagazineApi = injectAPIGroup({
    /**
     * 获取专题下的全部评论
     *
     * @param {number} magazineId
     * @returns
     */
    getNotes(magazineId) {
        return magazineNoteHttp.get(`/magazine/${magazineId}/note`);
    },
    /**
     *专题创建评论
     *
     * @param {number} magazineId
     * @param {NoteFormInterface} body
     * @returns
     */
    createNote(magazineId, body) {
        return NoteBaseHttp.post(`/magazine/${magazineId}/note`, body);
    },
    /**
     *更新评论
     *
     * @param {number} magazineId
     * @param {number} noteId
     * @param {NoteFormInterface} body
     * @returns
     */
    updateNote(magazineId, noteId, body) {
        return NoteBaseHttp.put(`/magazine/${magazineId}/note/${noteId}`, body);
    },
    /**
     *删除评论
     *
     * @param {number} magazineId
     * @param {number} noteId
     * @returns
     */
    deleteNote(magazineId, noteId) {
        return Http.delete(`/magazine/${magazineId}/note/${noteId}`);
    },
    /**
     *评论点赞
     *
     * @param {number} magazineId
     * @param {number} noteId
     * @returns
     */
    agreeNote(magazineId, noteId) {
        return Http.post(`/magazine/${magazineId}/note/${noteId}/agree`);
    },
    /**
     *取消评论点赞
     *
     * @param {number} magazineId
     * @param {number} noteId
     * @returns
     */
    disagreeNote(magazineId, noteId) {
        return Http.delete(`/magazine/${magazineId}/note/${noteId}/agree`);
    },
    /**
     *置顶评论
     *
     * @param {number} magazineId
     * @param {number} noteId
     * @param {{
     *       stick: boolean
     *     }} params
     * @returns
     */
    stickNote(magazineId, noteId, params) {
        return Http.post(`/magazine/${magazineId}/note/${noteId}/stick`, null, {
            params,
        });
    },
});
