// @tencent/mk-api/lib/define/base.js
import { pick } from './helper';
export const NumberDTO = 1;
export const StringDTO = '';
export const BooleanDTO = true;
export const AnyDTO = '';
export const BaseUser = {
    userId: NumberDTO,
    username: StringDTO,
};
export const BaseTime = {
    createdAt: NumberDTO,
    updatedAt: NumberDTO,
};
export const Rootable = {
    rootableId: NumberDTO,
    rootableType: StringDTO,
};
export const Linkable = {
    linkableId: NumberDTO,
    linkableType: StringDTO,
};
export const Agree = {
    agreeCount: NumberDTO,
    agreeUsernames: StringDTO,
};
export const DisAgree = {
    disagreeCount: NumberDTO,
    disagreeUsernames: StringDTO,
};
export const Like = {
    likeCount: NumberDTO,
    likeUsers: StringDTO,
};
export const Dislike = {
    dislikeCount: NumberDTO,
    dislikeUsers: StringDTO,
};
export const Content = {
    content: StringDTO,
    contentMd: StringDTO,
};
export const Title = {
    title: StringDTO,
    titleMd: StringDTO,
};
export const KV = {
    id: NumberDTO,
    name: StringDTO,
    value: StringDTO,
    orderBy: NumberDTO,
};
export const ShortKV = pick(KV, ['id', 'name']);
