// @tencent/mk-api/lib/api/answer/types.js
import { BooleanDTO, NumberDTO, StringDTO, Agree, DisAgree, Content, BaseTime, } from '../../define/base';
import { UserShortInfo } from '../user/types';
import { AnswerNote } from '../answerNote/types';
import { TypeWrapper } from '../../http';
import { extend, pick } from '../../define/helper';
export const Answer = extend({
    id: NumberDTO,
    problemId: NumberDTO,
    accepted: BooleanDTO,
    readCount: NumberDTO,
    noteCount: NumberDTO,
    imageField: StringDTO,
    videoField: StringDTO,
    contentPlain: StringDTO,
    author: UserShortInfo,
    source: StringDTO,
    post: StringDTO,
}, Agree, DisAgree, Content, BaseTime);
export const AnswerHttp = TypeWrapper(Answer);
export const AnswerWithNote = {
    answer: Answer,
    notes: AnswerNote,
};
export const AnswerWithNoteHttp = TypeWrapper(AnswerWithNote);
// 回答的精简信息, 一般用于questionList中
export const ShortAnswer = extend(pick(Answer, [
    'contentPlain',
    'contentMd',
    'problemId',
    'id',
    'agreeCount',
    'agreeUsernames',
    'accepted',
    'noteCount',
    'imageField',
    'videoField',
    'createdAt',
    'updatedAt',
]), {
    author: UserShortInfo,
});
