// @tencent/mk-api/lib/api/project/types.js
import { NumberDTO, StringDTO, BooleanDTO, BaseTime, ShortKV, } from '../../define/base';
import { TypeWrapper } from '../../http';
import { extend, pick } from '../../define/helper';
export const ProjectBase = extend({
    id: NumberDTO,
    sourceId: NumberDTO,
    sourceType: StringDTO,
    name: StringDTO,
    path: StringDTO,
    namespaceName: StringDTO,
    namespacePath: StringDTO,
    avatar: StringDTO,
    tags: [ShortKV],
    description: StringDTO,
    starCount: NumberDTO,
    mkStarCount: NumberDTO,
    viewCount: NumberDTO,
    lastActivityAt: NumberDTO,
}, BaseTime);
export const Project = extend({
    curStar: BooleanDTO,
}, ProjectBase);
export const ProjectHttp = TypeWrapper(Project);
export const ProjectShort = pick(Project, [
    'avatar',
    'description',
    'id',
    'name',
    'namespacePath',
    'path',
    'sourceId',
    'tags',
]);
export const ProjectShortHttp = TypeWrapper(ProjectShort);
export const IssuesBase = extend({
    title: StringDTO,
    iid: NumberDTO,
}, BaseTime);
export const Issues = extend({
    project: ProjectShort,
}, IssuesBase);
export const IssuesHttp = TypeWrapper(Issues);
